src/wwwroot/js/genpage/main.js

```diff
@@ -102,7 +102,26 @@ function rightClickImageInBatch(e, div) {
     if (e.shiftKey || e.ctrlKey) {
         return;
     }
-    let popover = new AdvancedPopover('image_batch_context_menu', [ { key: 'Remove', action: () => div.remove() } ], false, mouseX, mouseY, document.body, null);
+
+    let src = div.dataset.src;
+    let fullsrc = getFullsrc(src);
+    let metadata = div.dataset.metadata;
+    let buttonsChoice = getButtonsChoice();
```

Comment:
right click on batch has no reason to interact with the buttons-under-image choice